### PR TITLE
Add missing subtype ee cls

### PR DESCRIPTION
**Pull request by kiadso**
Adding checks for the other two subsidy-related data elements. Eva already had the one for Destination, but not for Prior Living Situation and Current Living Situation. These are added now, also I wrote a script that edits the sample data for the Exit, Enrollment, and CurrentLivingSituation csvs so that each of these three subsidy checks will flag 5 records for each issue. Built against the current-good ICF sample dataset, but there's code included that would help you find the proper records to flag if you were using a different dataset.

### CODE REVIEW

File: public-resources/EvaChecks.csv [hunk 1]
```diff
@@ -53,7 +53,8 @@ dq,Error,Incorrect Year Entered Service,V1.1 - Veteran's Information: Year Enter
...
 dq,Error,Invalid Move-In Date,3.20 - Housing Move-In Date,This move-in date does not fall between the Entry Date and the Exit Date or this move-in date is after the Export End Date. Please review and correct.,,40
 dq,Error,Missing Approximate Date Homeless,3.917.3 - Prior Living Situation: Date Homelessness Started,This data element is required to be collected at Project Start. Please go to the client's assessment at Project Start to enter this data to HMIS.,,28
-dq,Error,Missing Destination Subsidy Type,"3.12.1 - Destination, 3.12.A - Destination Subsidy Type","If a client exits to Rental by Client, the user must record the Subsidy Type.",,121
+dq,Error,Missing Current Living Situation Subsidy Type,"4.12 - Current Living Situation, 4.12A Current Living Situation Subsidy Type","If the response to Current Living Situation is ""Rental by client, with ongoing housing,"" then the Rental Subsidy Type must be specified.",,129
+dq,Error,Missing Destination Subsidy Type,"3.12.1 - Destination, 3.12.A - Destination Subsidy Type","If the response to Destination is ""Rental by client, with ongoing housing,"" then the Rental Subsidy Type must be specified.",,121
```
Comment (alex-silverman) [inline public-resources/EvaChecks.csv:57]: Isn't the value "Rental by client, with ongoing housing **subsidy**"?  And I think this should say that "the **_Destination_** Subsidy Type must be specified **and valid**"

File: public-resources/EvaChecks.csv [hunk 2]
```diff
@@ -53,7 +53,8 @@ dq,Error,Incorrect Year Entered Service,V1.1 - Veteran's Information: Year Enter
 dq,Error,Incorrect Year Separated,V1.1 - Veteran's Information: Year Separated from Military Service,The year the client separated from military service is in the future. Please verify the dates this client was in military service to correct this error.,,18
 dq,Error,Invalid Move-In Date,3.20 - Housing Move-In Date,This move-in date does not fall between the Entry Date and the Exit Date or this move-in date is after the Export End Date. Please review and correct.,,40
 dq,Error,Missing Approximate Date Homeless,3.917.3 - Prior Living Situation: Date Homelessness Started,This data element is required to be collected at Project Start. Please go to the client's assessment at Project Start to enter this data to HMIS.,,28
-dq,Error,Missing Destination Subsidy Type,"3.12.1 - Destination, 3.12.A - Destination Subsidy Type","If a client exits to Rental by Client, the user must record the Subsidy Type.",,121
+dq,Error,Missing Current Living Situation Subsidy Type,"4.12 - Current Living Situation, 4.12A Current Living Situation Subsidy Type","If the response to Current Living Situation is ""Rental by client, with ongoing housing,"" then the Rental Subsidy Type must be specified.",,129
```
Comment (alex-silverman) [inline public-resources/EvaChecks.csv:56]: Shouldn't this say "If the response to Current Living Situation is 'Rental by client, with ongoing housing **subsidy**"? Same with the others.

File: 05_DataQuality.R [hunk 3]
```diff
@@ -645,10 +645,35 @@ dkr_destination <- base_dq_data %>%
...
            Destination == 435 &
-           is.na(DestinationSubsidyType)) %>%
+           (is.na(DestinationSubsidyType) |
+           !DestinationSubsidyType %in% c(subsidy_types))) %>%
   merge_check_info(checkIDs = 121) %>%
   select(all_of(vars_we_want))
 
+# Missing ResPrior Subsidy ------------------------------------------------
+
+missing_res_prior_subsidy <- base_dq_data %>%
```
Comment (alex-silverman) [inline 05_DataQuality.R:655]: why "prior"?

File: public-resources/EvaChecks.csv [hunk 4]
```diff
@@ -67,6 +68,7 @@ dq,Error,Missing Months or Times Homeless,"3.917.4 - Prior Living Situation: Tim
 dq,Error,Missing Name Data Quality,3.01.5 - Name: Name Data Quality,Please correct by providing the missing information in the client's record.,,33
 dq,Error,Missing Percent AMI,V4 - Percent of AMI (SSVF Eligibility),This data element is required to be collected at Project Start. Please go to the client's assessment at Project Start to enter this data to HMIS.,,22
 dq,Error,"Missing Previously Unsheltered, ES, SH","3.917.2C - Prior Living Situation: Previous Street, ES, SH",This data element is required to be collected at Project Start. Please go to the client's assessment at Project Start to enter this data to HMIS.,,29
+dq,Error,Missing Prior Living Situation Subsidy Type,"3.917 - Prior Living Situation, 3.917A Rental Subsidy Type","If the response to Prior Living Situation is ""Rental by client, with ongoing housing,"" then the Rental Subsidy Type must be specified.",,130
```
Comment (alex-silverman) [inline public-resources/EvaChecks.csv:71]: ditto comment as above about adding "...ongoing housing **subsidy**"

File: sandbox/change-sample-data/add-missing-subsidies.R
Comment (alex-silverman) [inline sandbox/change-sample-data/add-missing-subsidies.R:1]: love this! I'll work on incorporating into testing

**Review by alex-silverman — COMMENTED** [review]
Left a few comments, mostly re: Checks language. Again, if you don't think the changes need to be made, go ahead and merge!

File: 05_DataQuality.R (hunk 3 above)
Comment (kiadso) [reply 05_DataQuality.R:655]: :) I think a long time ago, Enrollment.LivingSituation was called Enrollment.ResidencePrior? Either way, even now, that subset of data elements (see the screenshot) is called "Prior Living Situation." The meaning of LivingSituation in the Enrollment table is the place the household is coming from, or, their "residence prior" to entry into whatever project it is. As opposed to "CurrentLivingSituation." 
image

File: public-resources/EvaChecks.csv (hunk 1 above)
Comment (kiadso) [reply public-resources/EvaChecks.csv:57]: thanks, good call- I like the "...and valid" part since we are checking that

**Review by alex-silverman — APPROVED** [review]

**Pull request merged**